src/libs/compound-v3/service.ts

```diff
@@ -0,0 +1,348 @@
+        borrowCapacity = borrowCapacityValue
+          .div(baseTokenPrice)
+          .decimalPlaces(baseToken.decimals, BigNumberJS.ROUND_FLOOR)
+          .toFixed();
+
+        liquidationLimit = liquidationLimit.plus(collateralValue.times(liquidateCollateralFactor));
+      }
+
+      const collateralInfo: CollateralInfo = {
+        asset: asset.unwrapped,
+        assetPrice,
+        borrowCollateralFactor,
+        liquidateCollateralFactor,
+        collateralBalance,
+        collateralValue: formatValue(collateralValue),
+        borrowCapacity,
+        borrowCapacityValue: formatValue(borrowCapacityValue),
+      };
+
+      collaterals.push(collateralInfo);
+    }
+
+    let borrowCapacity = new BigNumberJS('0');
+    let availableToBorrow = new BigNumberJS('0');
+    let availableToBorrowValue = '0';
+    if (!totalBorrowCapacityValue.isZero()) {
+      borrowCapacity = totalBorrowCapacityValue
+        .div(baseTokenPrice)
+        .decimalPlaces(baseToken.decimals, BigNumberJS.ROUND_FLOOR);
+      availableToBorrow = borrowCapacity.minus(borrowBalance);
+      availableToBorrowValue = formatValue(availableToBorrow.times(baseTokenPrice));
+    }
+
+    let liquidationThreshold = '0';
+    let liquidationRisk = new BigNumberJS(0);
+    let liquidationPointValue = new BigNumberJS(0);
+    let liquidationPoint = '0';
+    if (!liquidationLimit.isZero()) {
+      liquidationThreshold = liquidationLimit.div(totalCollateralValue).decimalPlaces(4).toFixed();
+      liquidationRisk = new BigNumberJS(borrowValue).div(liquidationLimit).decimalPlaces(2);
+      liquidationPointValue = totalCollateralValue.times(liquidationRisk);
+      liquidationPoint = liquidationPointValue
+        .div(baseTokenPrice)
+        .decimalPlaces(baseToken.decimals, BigNumberJS.ROUND_FLOOR)
+        .toFixed();
+    }
+
+    const utilization = calcUtilization(totalBorrowCapacityValue, borrowValue);
+    const healthRate = calcHealthRate(supplyValue, totalCollateralValue, borrowValue, liquidationThreshold);
+    const netApr = calcNetApr(supplyValue, supplyApr, totalCollateralValue, borrowValue, borrowApr);
+
+    const marketInfo: MarketInfo = {
+      baseToken: baseToken.unwrapped,
+      baseTokenPrice,
+      supplyApr,
+      supplyBalance,
+      supplyValue,
+      borrowApr,
+      borrowBalance,
+      borrowValue,
+      collateralValue: formatValue(totalCollateralValue),
+      borrowCapacity: borrowCapacity.toFixed(),
+      borrowCapacityValue: formatValue(totalBorrowCapacityValue),
+      availableToBorrow: availableToBorrow.toFixed(),
+      availableToBorrowValue,
+      liquidationLimit: formatValue(liquidationLimit),
+      liquidationThreshold,
+      liquidationRisk: liquidationRisk.toFixed(),
+      liquidationPoint,
+      liquidationPointValue: formatValue(liquidationPointValue),
+      utilization,
+      healthRate,
+      netApr,
+      collaterals,
+    };
```

Comment:
Since this api serves user and non-user at the same time, maybe differentiate them from the naming will be easier to understand.

Reply:
I primarily reference the display of Compound v3 's homepage and the design structure of front end.
Because the calculation of data and user information presence are closely related, it would be better to define it in the PRD in this regard.